### ISSUE TITLE
Weight pruning in Keras for R

### ISSUE THREAD

**Issue by faltinl** [rstudio/keras#1150]
Weight pruning seems to be a valuable tool to improve neural networks' ability to generalize by throwing away those parts (weights) of the network which do not significantly contribute to convergence and accuracy. An introductory article can be found under [Obtaining Top Neural Network Performance Without Any Training](https://medium.com/analytics-vidhya/obtaining-top-neural-network-performance-without-any-training-5af0af464c59) and references given therein.

Unfortunately I cannot find any useful tools among the help pages of my Keras 2.3.0.0. package in R. Are there any other helpful packages or ready-to-use workarounds somewhere else to apply the technique of weight pruning (structured or unstructured) within Keras for R?

Any helpful hints would be highly appreciated.

**Comment by dfalbel** [comment]
Currently there's no plan to port `tensorflow_model_optimization` to R.

In theory you could install the library with: `reticulate::py_install("tensorflow-model-optimization", pip = TRUE)` and then use it with `tfmod <- reticulate::import("tensorflow_model_optimization")`.

**Comment by faltinl** [comment]
> Currently there's no plan to port tensorflow_model_optimization to R.

Oh, what a pity!

> In theory you could install the library with: reticulate::py_install("tensorflow-model-optimization", pip = TRUE) and then use it with tfmod <- reticulate::import("tensorflow_model_optimization").

Sounds quite challenging. Perhaps I shall try it... Thank you nevertheless, Daniel!
Cheers, Leo
